src/Component/Attribute/AsGrid.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Grid\Attribute;
+
+#[\Attribute(\Attribute::TARGET_CLASS)]
+final class AsGrid
+{
+    public function __construct(
+        public readonly string $resourceClass,
```

Comment:
It should be optional, cause grid can be unrelated to resource classes.

Reply:
Should we add test to configure a non resource related ?